app/src/main/java/com/oracle/javafx/scenebuilder/app/SceneBuilderApp.java

```diff
@@ -507,24 +527,81 @@ public void handleOpenFilesAction(List<String> files) {
         }
 
         EditorController.updateNextInitialDirectory(fileObjs.get(0));
-
+        
         // Fix for #45
         if (userLibrary.isFirstExplorationCompleted()) {
-            performOpenFiles(fileObjs);
+            var openResult = performOpenFiles(fileObjs);
+            handleFileOpenResult(openResult, onSuccess, WelcomeDialogWindowController.getInstance().getStage());
```

Comment:
These 2 steps are dependent on each other. I am wondering if we should merge them. This way we will be able to do:

```
performOpenFiles(fileObjs, onSuccess);
```

Reply:
Good catch. There is now a new method combining this.